src/libraries/System.Private.CoreLib/src/System/Enum.cs

```diff
@@ -1086,6 +1086,25 @@ public static string Format(Type enumType, object value, [StringSyntax(StringSyn
 
             throw new FormatException(SR.Format_InvalidEnumFormatSpecification);
         }
+
+        public static bool TryFormat<TEnum>(TEnum value, Span<char> destination, out int charsWritten) where TEnum : struct, Enum
+        {
+            if (!destination.IsEmpty)
+            {
+                string? enumName = InternalFormat((RuntimeType)typeof(TEnum), ToUInt64(value));
+
+                if (enumName != null
+                    && destination.Length >= enumName.Length)
+                {
+                    enumName.CopyTo(destination);
+                    charsWritten = enumName.Length;
+                    return true;
+                }
+            }
+
+            charsWritten = 0;
```

Comment:
The only reason a TryFormat method should return false is if the destination buffer is too small.

Reply:
This was my intention, I'll make sure the tests reflect that.

Reply:
Here it's also returning false if InternalFormat returns null.